src/modules/fancyzones/lib/ZoneWindow.cpp

```diff
@@ -522,13 +522,13 @@ void ZoneWindow::LoadSettings() noexcept
     JSONHelpers::FancyZonesDataInstance().AddDevice(m_uniqueId);
 }
 
-void ZoneWindow::InitializeZoneSets(MONITORINFO const& mi) noexcept
+void ZoneWindow::InitializeZoneSets(bool newWorkArea) noexcept
 {
     auto parent = m_host->GetParentZoneWindow(m_monitor);
     if (parent)

```

Comment:
Not strictly related to this PR, but this `parent` here is not really a parent, and also `GetParentZoneWIndow`is not thread safe. When I say it is not really a parent, I mean that it is more like a "previously active work area". Since we still have some use of it, maybe it is better to delete `GetParentZoneWindow` and simply pass this "previously active work area" unique identifier through `ZoneWindow` constructor.

Reply:
Let's do this as a follow up. I have some other similar things in mind

Reply:
Could You create a follow up ticket for this one, so we can keep track of it?

Reply:
Sure